src/main/java/org/carlmontrobotics/subsystems/Arm.java

```diff
@@ -26,23 +73,85 @@ public Arm() {
+      targetPosition = getArmClampedGoal(targetPosition);
 
+      
     }
-    
-    public void setSpeed() {
-      //move the arm around based off the right joystick movement on the manipulator joystick
-      //use the trapezoid thingy from robot code 2023
+
+    public void COMBINE_PID_FF_TRAPEZOID(TrapezoidProfile.State setPoint) {
+      // feed forward still needs the math part
+      double armFeedVolts = armFeed.calculate(setPoint.velocity, 0);
+      armPID.setReference(setPoint.position, CANSparkMax.ControlType.kPosition,0, armFeedVolts);
     }
-    
+
+    public double getArmPos() {
+      return MathUtil.inputModulus(armEncoder.getPosition(), Constants.Arm.ARM_DICONT_RAD,
+              Constants.Arm.ARM_DICONT_RAD + 2 * Math.PI);
+    } 
+
+
+    public void cancelArmCommand() {
+      Command currentArmCommand = getCurrentCommand();
+      currentArmCommand.cancel();
+    }
+
+    public Translation2d getCoM() {
+      // the constants are placeholders
+      Translation2d comOfArm = new Translation2d(Constants.Arm.COM_ARM_LENGTH_METERS, Rotation2d.fromRadians(getArmPos()))
+                .times(Constants.Arm.ARM_MASS_KG);
+      return comOfArm;
+    }
+
+    public double getArmVel() {
+      return armEncoder.getVelocity();
+    }
+
+    public void driveArm(TrapezoidProfile.State state) {
+      
+
+      /*
+      ignore this math its wrong as it includes wrist 
+        double kgv = getKg();
+        double armFeedVolts = kgv * getCoM().getAngle().getCos() + armFeed.calculate(state.velocity, 0);
+        double armPIDVolts = armPID.calculate(getArmPos(), state.position);
+        if ((getArmPos() > Constants.Arm.UPPER_ANGLE_LIMIT && state.velocity > 0) || 
+            (getArmPos() < Constants.Arm.LOWER_ANGLE_LIMIT && state.velocity < 0)) {
+            armFeedVolts = kgv * getCoM().getAngle().getCos() + armFeed.calculate(0, 0);
+    }
+    double volts = armFeedVolts + armPIDVolts;
+    armMotor1.setVoltage(volts);
+      
+  
+  */
+      double armFeedVolts = 0; //<-- similar math to above to get this
+      double armPIDVolts = 0;//<--similar math to 
+      SmartDashboard.putNumber("ArmFeedVolts", armFeedVolts);
+      SmartDashboard.putNumber("ArmPIDVolts", armPIDVolts);
+      double volts = armFeedVolts + armPIDVolts;
+      SmartDashboard.putNumber("ArmTotalVolts", volts);
+      armMotor1.setVoltage(volts);
+    }
+    public double getArmClampedGoal(double goal) {
+      //Find the limits of the arm. Used to move it and ensure that the arm does not move past the amount
+      return MathUtil.clamp(MathUtil.inputModulus(goal, Constants.Arm.ARM_DICONT_RAD, Constants.Arm.ARM_DICONT_RAD + 2 * Math.PI), Constants.Arm.LOWER_ANGLE_LIMIT, Constants.Arm.UPPER_ANGLE_LIMIT);
+    }
+
+    public TrapezoidProfile.State getCurrentArmState() {
+      return new TrapezoidProfile.State(getArmPos(), getArmVel());
+    }
+
+
     @Override
     public void periodic() {
```

Comment:
fill this out!
see `implemented-outtake` or RobotCode2024's `arm` periodic, but basically this should be doing what all your `armAmpPos`/`armGroundPos`/etc commands do.